tests/js/percentageTests.js

```diff
@@ -0,0 +1,129 @@
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 4.4, 4.4, 0, 0, 100, 100, 200, 200], // total === "50"
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 2.2, 2.2, 0, 0, 50, 50, 100, 100], // total === 100
+        [undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, undefined, 2.2, 2.2, 0, 0, 50, 50, 100, 100] // total === "100"
+    ];
+
+    jqUnit.test("Test gpii.chartAuthoring.percentage.calculate", function () {
+        fluid.each(gpii.tests.calculatePercentage.inputs, function (total, totalIdx) {
+            fluid.each(gpii.tests.calculatePercentage.inputs, function (value, valIdx) {
+                var actual = gpii.chartAuthoring.percentage.calculate(value, total);
+                var expected = gpii.tests.calculatePercentage.outputs[totalIdx][valIdx];
+                // Using fluid.model.isSameValue to remove any javascript precision errors on floating point numbers.
+                jqUnit.assertTrue("The percentate for value: " + value + " total: " + total + " should be calculated as " + expected, fluid.model.isSameValue(expected, actual));
+            });
+        });
+    });
+
+    fluid.registerNamespace("gpii.tests.retrieveVal");
+
+    gpii.tests.retrieveVal.percentages = [
+        "10.5",
+        10.5,
+        "10",
+        10,
+        function () {return "10.5";},
+        function () {return 10.5;},
+        function () {return "10";},
+        function () {return 10;}
+    ];
+
+    gpii.tests.retrieveVal.percentages = ["10.5", 10.5, "10", 10, "10.5", 10.5, "10", 10];
+
+    jqUnit.test("Test gpii.chartAuthoring.percentage.retrieveVal", function () {
+        fluid.each(gpii.tests.retrieveVal.percentages, function (percentage, idx) {
+            jqUnit.assertEquals("The value should have been retrieved correctly", gpii.tests.retrieveVal.percentages[idx], gpii.chartAuthoring.percentage.retrieveVal(percentage));
+        });
+    });
+
+    fluid.registerNamespace("gpii.tests.renderPerencentage");
+
+    gpii.tests.renderPerencentage.templates = [undefined, "%percentage%"];
+
+    gpii.tests.renderPerencentage.outputs = [
+        ["10.5", "10.5", "10", "10", "10.5", "10.5", "10", "10"], // template === undefined
+        ["10.5%", "10.5%", "10%", "10%", "10.5%", "10.5%", "10%", "10%"] // template === "%percentage%"
+    ];
+
+    jqUnit.test("Test gpii.chartAuthoring.percentage.render", function () {
+        var elm = $(".renderPerencentage-test");
+        fluid.each(gpii.tests.renderPerencentage.templates, function (template, templateIdx) {
+            fluid.each(gpii.tests.retrieveVal.percentages, function (percentage, perIdx) {
+                // make sure the text is reset
+                elm.text("");
```

Comment:
Cleaner to issue a fresh jqUnit test on each round of the loop, and rely on QUnit's native markup cleansing rather than do this by hand